core/src/components/datetime/datetime.tsx

```diff
@@ -408,6 +415,13 @@ export class Datetime implements ComponentInterface {
    */
   @Event() ionStyle!: EventEmitter<StyleEventDetail>;
 
+  /**
+   * Emitted when the working parts change. Used for testing
+   * purposes, e.g. to ensure month selection works.
+   * @internal
+   */
+  @Event() ionWorkingPartsDidChange!: EventEmitter<DatetimeParts>;
```

Comment:
This should replace the `InitMonthDidChangeEvent` function completely. I'll make a separate PR to update the other tests.

Reply:
I wasn't able to find one, but I'm open to ideas. The old util (`InitMonthDidChangeEvent`) fits the bill and uses a mutation observer to indirectly watch for the month to change, but it doesn't account for the month element being re-rendered (and thus the reference to the observed element being lost) when `presentation` is changed.

Reply:
Spent some time exploring an alternative way to test this behavior. Here's a branch that shows how we can drop the internal event from `ion-datetime`: https://github.com/ionic-team/ionic-framework/tree/test/FW-844.

It will apply a mutation observer to dispatch a custom event once the calendar body is manipulated. It will also re-attach the observer to the correct element node after the presentation is changed. 

If you like this approach, feel free to take any parts of it/merge it into your working branch.

Reply:
@sean-perkins Thanks for that! I cleaned it up and brought it in 👍 cc @liamdebeasi